js/foundation.offcanvas.js

```diff
@@ -157,6 +172,18 @@ class OffCanvas extends Plugin {
       var $target = this.options.contentOverlay ? this.$overlay : this.$content;
       $target.on({'click.zf.offcanvas': this.close.bind(this)});
     }
+
+    if (this.options.inCanvasOn) {
+      $(window).on('changed.zf.mediaquery', () => {
+        if(MediaQuery.atLeast(this.options.inCanvasOn)) {
+          this.close();
+          this.isInCanvas = true;
+        } else {
+          this.isInCanvas = false;
+        }
```

Comment:
Basically:
```js
if (true) {
  true;
} else {
  false;
}
```
Can be factorized ;)

Reply:
Yup didn't notice this while writing it. It's refactored in `_checkInCanvas` now.